agent-inject/handler.go

```diff
@@ -25,6 +26,8 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+const warningInitOnlyInjection = "workload uses initContainer only; consider migrating to VSO"
```

Comment:
Feels like this could get noisy for folks who aren't looking to move to VSO just yet? And workloads other than init-only are certainly suitable for VSO too. I think I'd prefer to leave this sort of warning out, please.